sqlglot/tokens.py

```diff
@@ -843,8 +843,16 @@ def tokenize(self, sql: str) -> t.List[Token]:
         self.reset()
         self.sql = sql
         self.size = len(sql)
-
         try:
+            # mysql dumps partition details as conditional commands. The command starts with the same version number always, the below code is used to make it compatible with NuoDB
+            while "!50100" in self.sql:
```

Comment:
Are we going to have to keep updating this condition?  Seems strangely specific.  Is there a regex we can use instead?

Reply:
I have updated the IF condition. Please let me know what you feel about it